proknow/Audit.py

```diff
@@ -0,0 +1,130 @@
+__all__ = [
+    'Audit',
+]
+
+import json
+import datetime
+
+from proknow.Exceptions import InvalidOperationError
+
+class Audit(object):
+    """
+    This class should be used to interact with the audit logs in a Proknow organization.
+    """
+
+    def __init__(self, proknow, requestor):
+        """Initializes the Audit class.
+
+        Parameters:
+            proknow (proknow.ProKnow.ProKnow): The ProKnow instance that is instantiating the
+                object.
+            requestor (proknow.Requestor.Requestor): An object used to make API requests.
+        """
+        self._proknow = proknow
+        self._requestor = requestor
+        self.auditResultsPage = None
+    
+    def query(self, page_size=25, start_time=None, end_time=None,
```

Comment:
Handle all of the following possible query parameters:

```javascript
"query": joi.object().keys({
            "page_number": joi.when('first_id', {
                is: joi.exist(),
                then: joi.number().integer().min(0).required(),
                otherwise: joi.forbidden()
            }),
            "page_size": joi.number().integer().min(1).max(200),
            "first_id": joi.string().regex(utils.idRegex),
            "start_time": joi.string().isoDate(),
            "end_time": joi.string().isoDate(),
            "types": joi.array().items(joi.string().max(255)),
            "user_id": joi.string().regex(utils.idRegex),
            "user_name": joi.string().max(255),
            "patient_id": joi.string().regex(utils.idRegex),
            "patient_name": joi.string().max(255),
            "patient_mrn": joi.string().max(255),
            "workspace_id": joi.string().regex(utils.idRegex),
            "workspace_name": joi.string().max(255),
            "collection_id": joi.string().regex(utils.idRegex),
            "resource_id": joi.string().max(32).allow(null),
            "resource_name": joi.string().max(255),
            "classification": joi.string().valid(['HTTP', 'AUTH']),
            "methods": joi.array().items(joi.string().valid(['GET', 'HEAD', 'POST', 'PUT', 'DELETE',
                'CONNECT', 'OPTIONS', 'TRACE', 'PATCH'
            ])),
            "uri": joi.string().max(1000),
            "user_agent": joi.string().max(255),
            "ip_address": joi.string().max(39),
            "status_codes": joi.array().items(joi.string().max(3).regex(/^[0-9]+$/)),
            "text": joi.array().items(joi.string()).max(50)
        }),
```